src/main/java/com/github/mauricioaniche/ck/metric/TightClassCohesion.java

```diff
@@ -0,0 +1,128 @@
+        }
+        for(ImmutablePair<String, String> pair : directConnections){
+            directConnectionsMap.get(pair.left).add(pair.right);
+        }
+
+        //extract all direct and indirect connections between methods from the direct connections
+        HashMap<String, Set<String>> indirectConnectionsMap = new HashMap<>();
+        for (CKMethodResult method : result.getVisibleMethods()){
+            Set<String> localConnections = extractConnections(method.getMethodName(), new HashSet<>(), directConnectionsMap);
+            indirectConnectionsMap.put(method.getMethodName(), localConnections);
+        }
+
+        //map the indirect connections into connection pairs
+        Set<ImmutablePair<String, String>> indirectConnections = new HashSet<>();
+        for(String key : indirectConnectionsMap.keySet()){
+            indirectConnections.addAll(indirectConnectionsMap.get(key).stream()
+                    .filter(right -> !key.equals(right))
+                    .map(right -> new ImmutablePair<String, String>(key, right))
+                    .collect(Collectors.toSet()));
+        }
+        //remove all direct connections
+        indirectConnections.removeAll(directConnections);
+        return indirectConnections;
+    }
+
+    //Recursively extract all indirect connections between methods starting with the given method
+    //Explored contains all previously explored connections
+    //connections contains all direct method connections of interest
+    private Set<String> extractConnections(String currentConnection, Set<String> explored, HashMap<String, Set<String>> connections){
+        explored.add(currentConnection);
+
+        //only explore connections that were not previously explored
+        Set<String> nextConnections = connections.get(currentConnection).stream()
+                .filter(connection -> !explored.contains(connection))
+                .collect(Collectors.toSet());
+        explored.addAll(nextConnections);
+        for (String nextConnection : nextConnections){
+            explored.addAll(extractConnections(nextConnection, explored, connections));
+        }
+
+        //Stops when all invocations are explored: there are no more invocations to be explored
+        return explored;
+    }
+
+    public void setResult(CKClassResult result) {
+        MethodInvocationsLocal methodInvocationsLocal = new MethodInvocationsLocal();
+        methodInvocationsLocal.extractInvocations(result);
+
+        //maximum number of possible connections (N * (N -1))
+        float np = result.getVisibleMethods().size() * (result.getVisibleMethods().size() -1);
+
+        //number of direct connections (number of edges in the connection graph) in this class
+        Set<ImmutablePair<String, String>> directConnections = getDirectConnections(result);
+        result.setTightClassCohesion(directConnections.size() / np);
```

Comment:
Can't you have a division by zero here?

np could be zero if visible methods == 1, so `float np = 1 * 0`.

Reply:
that'd deserve a test, if you find that my assumption is true!

Reply:
Yes, you are right. I will treat this differently though. If the number of visible methods in a class is below 2, there is no reasonable value for TCC and LCC, because there are can be no connections between methods. Thus, I will set it to -1 in these cases.